game_server_launcher/launcher.py

```diff
@@ -328,7 +328,7 @@ def handle_match_time_message(self, msg):
             else:
                 time_left = self.min_next_switch_time - datetime.datetime.utcnow()
                 if time_left < datetime.timedelta(0):
-                    time_left = datetime.timedelta(seconds=1)
+                    time_left = datetime.timedelta(seconds=3)
```

Comment:
What was this bug?

Reply:
Oh, I don't think that strictly matters. I had thought there was a race condition there but the behaviour was explained by something unrelated.

Reply:
I was thinking about the delay of 3 seconds and if it would make sense to a reader. First I thought 1 second made more sense than 3, but then I realized that doing it right away is even better (if no time left then set otherwise schedule set). I could also live with 0 seconds if spawn_later accepts that and you think it looks better. What do you think?

Reply:
I guess it could be done straight away. Actually this branch should never happen at all, right? Because we should have gone into the other branch if there was no time left before the min_next_switch_time was up.

The only circumstance in which this would happen is if the timer finished between the evaluation of the outer condition and here.

I'm having it set the server as ready immediately, I guess it avoids a very, very small race condition.

Reply:
I have another suggestion:
```
if self.min_next_switch_time:
    time_left = (self.min_next_switch_time - datetime.datetime.utcnow()).total_seconds()
else:
    time_left = 0

if time_left > 0:
    self.pending_callbacks.add(self, time_left, self.set_server_ready)
else:
    self.set_server_ready()
```
It avoids duplication and separates the remaining time calculation from the decision to use a callback or call directly.

Reply:
That makes sense, changing it to this.